pkg/config/config.go

```diff
@@ -61,91 +61,53 @@ type Config struct {
 	EvalContext *hcl.EvalContext
 }
 
-func loadLokocfgPaths(configPath string) ([]string, error) {
-	isDir, err := util.PathIsDir(configPath)
-	if err != nil {
-		return nil, fmt.Errorf("failed to stat config path %q: %w", configPath, err)
-	}
-	var lokocfgPaths []string
-	if isDir {
-		globPattern := filepath.Join(configPath, "*.lokocfg")
-		configFiles, err := filepath.Glob(globPattern)
-		if err != nil {
-			return nil, fmt.Errorf("bad filepath glob pattern %q: %w", globPattern, err)
-		}
-		lokocfgPaths = append(lokocfgPaths, configFiles...)
-	} else {
-		lokocfgPaths = append(lokocfgPaths, configPath)
-	}
-	return lokocfgPaths, nil
-}
-
-func LoadConfig(lokocfgPath, lokocfgVarsPath string) (*Config, hcl.Diagnostics) {
-	lokocfgPaths, err := loadLokocfgPaths(lokocfgPath)
-	if err != nil {
-		return nil, hcl.Diagnostics{
-			&hcl.Diagnostic{
-				Severity: hcl.DiagError,
-				Summary:  err.Error(),
-			},
+// Component finds a component named name and returns its configuration as a pointer to an
+// hcl.Body. If the component isn't found, nil is returned.
+func (c *Config) Component(name string) *hcl.Body {
+	for _, component := range c.RootConfig.Components {
+		if name == component.Name {
+			return &component.Config
 		}
 	}
 
-	hclParser := hclparse.NewParser()
+	return nil
+}
 
-	var hclFiles []*hcl.File
-	for _, f := range lokocfgPaths {
-		hclFile, diags := hclParser.ParseHCLFile(f)
-		if len(diags) > 0 {
-			return nil, diags
-		}
-		hclFiles = append(hclFiles, hclFile)
+// ReadHCL reads HCL files at path and HCL variable files at varPath, constructs a Config and
+// returns a pointer to it.
+func ReadHCL(path, varPath string) (*Config, hcl.Diagnostics) {
+	body, diags := bodyFromPath(path)
+	if len(diags) > 0 {
+		return nil, diags
 	}
 
-	configBody := hcl.MergeFiles(hclFiles)
-
-	exists, err := util.PathExists(lokocfgVarsPath)
+	varsExist, err := util.PathExists(varPath)
 	if err != nil {
 		return nil, hcl.Diagnostics{
 			&hcl.Diagnostic{
 				Severity: hcl.DiagError,
-				Summary:  fmt.Sprintf("could not stat %q: %v", lokocfgVarsPath, err),
+				Summary:  fmt.Sprintf("could not stat %q: %v", varPath, err),
 			},
 		}
 	}
+
 	var userVals map[string]cty.Value
-	var diags hcl.Diagnostics
-	if exists {
-		userVals, diags = LoadValuesFile(lokocfgVarsPath)
+	if varsExist {
+		userVals, diags = readValues(varPath)
```

Comment:
I've changed `readValues` to `readValuesFile`. AFAICT both of these names are better than `LoadValuesFile` since "load" could mean many things but "read" typically refers to copying data from some source (a file in this case) to memory.

`populateVars()` is a function I've broken down from `ReadHCL()` following your [feedback](https://github.com/kinvolk/lokomotive/pull/806#discussion_r470735192). AFAICT its name is accurate since "populate vars" is exactly what this function does: it accepts a set of variables and a set of user-provided values and attempts to populate each variable with the corresponding value or with a default.

Is this clearer? Do you have better names in mind?

Reply:
`readValuesFile` sounds indeed better :+1: